prplmesh.bb: build RDKB (OE 2.2) image for RPi 3

MuteSpirit opened the pull request:
Add support to build prplmesh for TARGET_PLATFORM=linux and
BWL_TYPE=NL80211 to be able build RDKB (OE 2.2) image for Raspberry Pi 3 B+

Add patch to fix search headers and link for nl and wpa supplicant libraries

Replace ZeroMG with NNG because ZeroMG is licensed under LGPLv3 which
is deny in RDKB.

Add libnl bbappend to install CMake find package module for it

Has been used RDKB based on OE 2.2 - branch "rdkb-next"

All patches for prplMesh has been suggested for upstream project:
* https://github.com/prplfoundation/prplMesh/pull/1363
* https://github.com/prplfoundation/prplMesh/pull/1357
* https://github.com/prplfoundation/prplMesh/pull/1356
* https://github.com/prplfoundation/prplMesh/pull/1355

Closes https://jira.prplfoundation.org/browse/PPM-241

nmerinov commented on recipes-core/images/rdk-generic-broadband-image.bbappend:
```diff
@@ -0,0 +1,6 @@
+# SPDX-License-Identifier: BSD-2-Clause-Patent
```
I think you should add `BB_DANGLINGAPPENDS_WARNONLY` configuration as well to keep the possibility to use this layer with poky and without rdk-b

MuteSpirit replied:
`BB_DANGLINGAPPENDS_WARNONLY` has been added into 'conf/layer.conf'

VladyslavTupikin commented on conf/layer.conf:
```diff
@@ -14,4 +14,6 @@ BBFILE_COLLECTIONS += "prplmesh"
 BBFILE_PRIORITY_prplmesh = "9"
 
-LICENSE_PATH += "${LAYERDIR}/LICENSES"
+LICENSE_PATH += "${LAYERDIR}"
```
Why LICENSE_PATH it is LAYERDIR instead of LICENSES?

MuteSpirit replied:
Because `LIC_FILE_CHKSUM` contains filename `LICENSE` which is located in root of repo

arnout replied:
See my comment there.

MuteSpirit replied:
Solved in #19

VladyslavTupikin commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,123 @@ inherit cmake pythonnative
+BWL_TYPE ?= "bwl-nl80211"
 INSTALL_PREFIX ?= "/opt/prplmesh"
+#
+# QEMU MACHINE specific
```
I think we need to add a condition, which will check DISTRO_FEATURES and enable variables for qemu if qemu present in DISTRO_FEATURES.

MuteSpirit replied:
Explicit condition is not needed.
I'm using override syntax with MACHINE name (qemux86, qemuarm, etc.)
As a result variables with such suffix (_qemux86, etc.) will be used only if `${MACHINE}` is equal to `qemux86`. etc.

VladyslavTupikin commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,123 @@ inherit cmake pythonnative
+BWL_TYPE_qemux86 ?= "bwl-dummy"
+BWL_TYPE_qemux86-64 ?= "bwl-dummy"
+#
+# RPi machine specific
```
I think we need to add a condition, which will check DISTRO_FEATURES and enable variables for RPI if RPI present in DISTRO_FEATURES.

MuteSpirit replied:
See my answer for you comment to line
```
# QEMU MACHINE specific
```

nmerinov commented on conf/layer.conf:
```diff
@@ -14,4 +14,6 @@ BBFILE_COLLECTIONS += "prplmesh"
 BBFILE_PRIORITY_prplmesh = "9"
 
-LICENSE_PATH += "${LAYERDIR}/LICENSES"
+LICENSE_PATH += "${LAYERDIR}"
```
Why? `LICENSE_PATH` is a list of directories with `LICENSE` files. e.g. when you write `LICENSE = "MIT"` in recipes then yocto search for the file named "MIT" in this directories. 

If you keep your COPYING file on the top level of the layer this does not mean that you should point `LICENSE_PATH` to this directory.

MuteSpirit replied:
Solved in #19

nmerinov commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,123 @@ inherit cmake pythonnative
 
+SUMMARY = "prplMesh"
 SECTION = "net"
 LICENSE = "BSD-2-Clause-Patent.txt"
```
It's uncommon practice to have `.txt` in name of the license. Is it correct?

MuteSpirit replied:
@arnout add license file with `.txt` extension when created `meta-prplmesh` repo.
I'll ask him allow to rename license file

arnout replied:
The files in the LICENSES directory of prplMesh were created with the REUSE tool, so I guess they're correctly named...

nmerinov replied:
REUSE tool is not Yocto-style one. Common Yocto solution is the following: 

1. Name reference license file as in SPDX short identifier. Without any extensions. e.g. as in https://github.com/openembedded/openembedded-core/tree/master/meta/files/common-licenses
2. If there-is some well-known non-SPDX names for license files then aliases added with `SPDXLICENSEMAP` variable flags as in https://github.com/openembedded/openembedded-core/blob/master/meta/conf/licenses.conf

So I highly recommend to rename this file and remove `.txt` from name

arnout replied:
We're not going to rename the file in prplmesh just because yocto has a different convention.

However, mainly you mean that we don't need to use the license file from the prplmesh project, but can instead use one from the common licenses?

MuteSpirit replied:
Solved in #17

nmerinov commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,123 @@ inherit cmake pythonnative
+LIC_FILES_CHKSUM = "file://LICENSE;md5=0518d409dae93098cca8dfa932f3ab1b"
+
+DEPENDS = "python-native python-pyyaml-native json-c openssl"
 RDEPENDS_${PN} = "iproute2 busybox readline"
```
`RDEPENDS_${PN}` used for runtime dependencies. In Yocto-community this variable mostly used for dependencies from plugin to main application or for dependency between applications (e.g. called with "system()" call). It's uncommon to add shared library to RDEPENDS explicitely, because all required shared libraries can be found with `shdeps` mechanism

MuteSpirit replied:
Done

nmerinov commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,123 @@ inherit cmake pythonnative
-BUILD_TYPE ?= "Release"
+PACKAGECONFIG ?= "${BWL_TYPE} ${MSGLIB} ${BUILD_TYPE}"
+
+BUILD_TYPE ?= "release"
```
Why? https://cmake.org/cmake/help/v3.0/variable/CMAKE_BUILD_TYPE.html -- they all Capital Case types. If you want to configure build type with `PACKAGECONFIG` mechanism then it'll be better to remove this variable at all and define `PACKAGECONFIG ??= "${BWL_TYPE} mq-zmq release"`.
NOTE: I used BWL_TYPE through variable because you have pre-definition for different machines.

MuteSpirit replied:
Made `release` as default mode according your advice

VladyslavTupikin commented on recipes-connectivity/prplmesh/prplmesh/004-get_hostapd_ctrl_iface_pathes_from_conf.patch:
```diff
@@ -0,0 +1,286 @@
+ 
+     // Create a new AP HAL instance
+     ap_wlan_hal = bwl::ap_wlan_hal_create(
+diff --git a/agent/src/beerocks/slave/ap_manager_thread.h b/agent/src/beerocks/slave/ap_manager_thread.h
```
It is a bad idea do patch on the source code. Can you please move changes for source code to the main prplmesh project?

MuteSpirit replied:
There are prplmesh PRs for each of patch presented in `meta-prplmesh`
See links in this PR description

VladyslavTupikin commented on recipes-core/packagegroups/packagegroup-prplmesh.bb:
```diff
@@ -6,7 +6,9 @@
 
 LICENSE = "BSD-2-Clause-Patent.txt"
-LIC_FILES_CHKSUM = "file://LICENSE;md5=7b45146d47e73bcbac1068e5cfc2a9fb"
+# LIC_FILES_CHKSUM = "file://LICENSE;md5=7b45146d47e73bcbac1068e5cfc2a9fb"
```
Why string commented instead of removing?

MuteSpirit replied:
Solved in #19

VladyslavTupikin commented on recipes-core/packagegroups/packagegroup-prplmesh.bb:
```diff
@@ -6,7 +6,9 @@
 LICENSE = "BSD-2-Clause-Patent.txt"
-LIC_FILES_CHKSUM = "file://LICENSE;md5=7b45146d47e73bcbac1068e5cfc2a9fb"
+# LIC_FILES_CHKSUM = "file://LICENSE;md5=7b45146d47e73bcbac1068e5cfc2a9fb"
+# LIC_FILES_CHKSUM = "file://BSD-2-Clause-Patent.txt;md5=0518d409dae93098cca8dfa932f3ab1b"
```
Why string commented instead of removing?

MuteSpirit replied:
Solved in #19

VladyslavTupikin commented on recipes-support/libnl/libnl/FindLibNL.cmake:
```diff
@@ -0,0 +1,38 @@
+message("Searching for Linux netlink library")
```
Maybe better move *.cmake files to the main prplmesh repository?

MuteSpirit replied:
I remove added `*.cmake` files and use already exist in prplMesh

nmerinov commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,123 @@ inherit cmake pythonnative
+#
+# "wlan2" is not activate on RPi. So it's needed to plug-in Wi-Fi
+# USB dongle to have two wlan ifaces: wlan0 and wlan1
+BEEROCKS_WLAN1_IFACE_raspberrypi-rdk-broadband = "wlan0"
+BEEROCKS_WLAN2_IFACE_raspberrypi-rdk-broadband = "wlan1"
+BEEROCKS_HOSTAP_WLAN1_CTRL_IFACE_raspberrypi-rdk-broadband="/var/run/hostapd0/wlan0"
+BEEROCKS_HOSTAP_WLAN2_CTRL_IFACE_raspberrypi-rdk-broadband="/var/run/hostapd4/wlan1"
+BEEROCKS_BH_WIRE_IFACE_raspberrypi-rdk-broadband = "erouter0"
+
+PACKAGECONFIG_append_raspberrypi-rdk-broadband = " \
+    wlan1_iface \
+    wlan2_iface \
+    hostap_wlan1_ctrl_iface \
+    hostap_wlan2_ctrl_iface \
+    bh_wire_iface"
```
Do not looks like sane design: You require to define this variables twice

MuteSpirit replied:
Yes, i know
Firstly i used another variant:
```
EXTRA_OECMAKE = "
                .....................
                 ${@'-DBEEROCKS_BRIDGE_IFACE=${BEEROCKS_BRIDGE_IFACE}' if d.getVar('BEEROCKS_BRIDGE_IFACE', True) or ''} \
                ....... and similar lines for another BEEROCKS vars .......
                 "
```
But then rewrite to `PACKAGECONFIG` usage.

I think to keep in `PACKAGECONFIG` the variables which require changes in `DEPENDS` when turned on/off and the rest ones - add into 'EXTRA_OECMAKE' similar the example above

MuteSpirit replied:
Done

nmerinov commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,123 @@ inherit cmake pythonnative
+#                 -DCMAKE_LIBRARY_PATH=${STAGING_LIBDIR}
+#                 -DCMAKE_SKIP_RPATH:BOOL=ON
+
+PACKAGECONFIG[release] = "-DCMAKE_BUILD_TYPE=Release,,,"
+PACKAGECONFIG[debug] = "-DCMAKE_BUILD_TYPE=Debug,,,"
```
Looks like place with alternative design like: `PACKAGECONFIG[debug] = "-DCMAKE_BUILD_TYPE=Debug,-DCMAKE_BUILD_TYPE=Release,,"`

nmerinov commented on recipes-support/libnl/libnl_%.bbappend:
```diff
@@ -0,0 +1,11 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
+
+SRC_URI_append = " file://FindNL.cmake \
+                   file://FindLibNL.cmake \
+                 "
+
+do_install_append() {
+    install -d ${D}${datadir}/cmake/Modules
+    # install -m 0644 ${WORKDIR}/FindNL.cmake ${D}${datadir}/cmake/Modules
+    install -m 0644 ${WORKDIR}/FindLibNL.cmake ${D}${datadir}/cmake/Modules
```
What the reason to add two cmake modules manually and install only one of them? Why this modules added to libnl? Libnl already provided pkg-config that can define all variables for all build systems

arnout replied:
In addition, it's not going to be used by anything... prplMesh does `find_package(nl-3)` so it will load a file called `Findnl-3.cmake`, not `FindNL.cmake`.

This commit really should be split into smaller commits (that preferably still work with current OE) where each commit message explain what problem is solved.

MuteSpirit replied:
This bbappend has been removed
`prplmesh.bb` started use `*.cmake` files from prplMesh sources tree

arnout commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -10,25 +10,43 @@ OECMAKE_SOURCEPATH = "${S}/"
 LICENSE = "BSD-2-Clause-Patent.txt"
-LIC_FILES_CHKSUM = "file://LICENSE;md5=7b45146d47e73bcbac1068e5cfc2a9fb"
-DEPENDS = "python-native python-pyyaml-native json-c openssl zeromq"
+# LIC_FILES_CHKSUM = "file://LICENSE;md5=7b45146d47e73bcbac1068e5cfc2a9fb"
```
Why comment this out? BTW how did you get this hash to begin with? The LICENSE file has had the 0518 hash since half of April.

nmerinov replied:
This is because initial recipe was made on prplfoundation/prplMesh@911392026280d9961823a9359e6cc4f08c9ee0a2 commit

arnout commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -10,25 +10,43 @@ OECMAKE_SOURCEPATH = "${S}/"
-LIC_FILES_CHKSUM = "file://LICENSE;md5=7b45146d47e73bcbac1068e5cfc2a9fb"
-DEPENDS = "python-native python-pyyaml-native json-c openssl zeromq"
+# LIC_FILES_CHKSUM = "file://LICENSE;md5=7b45146d47e73bcbac1068e5cfc2a9fb"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=0518d409dae93098cca8dfa932f3ab1b"
```
I guess you changed the LICENSES_PATH because the LICENSE file is in the top directory, not in the LICENSES subdirectory. So I think this should be
```suggestion
LIC_FILES_CHKSUM = "file://BSD-2-Clause-Patent.txt;md5=0518d409dae93098cca8dfa932f3ab1b"
```

nmerinov replied:
This line is about license file in the https://github.com/prplfoundation/prplMesh/tree/d9eb170b69c1c225e40df5c191e8978568ec2e08 repository, not about reference license file in `LICENSES_PATH`

MuteSpirit replied:
Solved in #19

arnout commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -10,25 +10,43 @@ OECMAKE_SOURCEPATH = "${S}/"
                  -DCMAKE_INSTALL_PREFIX:PATH=${INSTALL_PREFIX} \
                  -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
-                 -DMSGLIB=zmq"
+                 -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
```
This is a bit weird to have in a production build...

nmerinov replied:
Actually maximal verbosity is sane configuration for build systems like Yocto: By default all logs hided by yocto itself, but in case of issues verbose build allow to get information what exactly goes wrong.

arnout commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -10,25 +10,43 @@ OECMAKE_SOURCEPATH = "${S}/"
+                 -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+                 -DCMAKE_INCLUDE_PATH=${STAGING_INCDIR} \
+                 -DCMAKE_LIBRARY_PATH=${STAGING_LIBDIR} \
+                 -DMSGLIB=nng"
```
The MSGLIB option will be removed in a week or two, so be prepared to get rid of this again :-)

MuteSpirit replied:
Thanks for news
Recipe of this PR uses strict commit so not depend on this changes.
I'll support removing `MSGLIB` in separate task

arnout commented on recipes-core/packagegroups/packagegroup-prplmesh.bb:
```diff
@@ -6,7 +6,9 @@
 LICENSE = "BSD-2-Clause-Patent.txt"
-LIC_FILES_CHKSUM = "file://LICENSE;md5=7b45146d47e73bcbac1068e5cfc2a9fb"
+# LIC_FILES_CHKSUM = "file://LICENSE;md5=7b45146d47e73bcbac1068e5cfc2a9fb"
+# LIC_FILES_CHKSUM = "file://BSD-2-Clause-Patent.txt;md5=0518d409dae93098cca8dfa932f3ab1b"
```
So I believe this one should be correct with the current _PATH.

Commit:
layer.conf: add Yocto flag to ignore absense of recipe if bbappend is present

This meta layer will be used both for Yocto Poky and RDKB.
And it'll be contain bbappend files for both.

Add BB_DANGLINGAPPENDS_WARNONLY into layer.conf
to avoid add more complex code for hide some of them from bitbake

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
nng: add nanomsg-next-generation recipe to use it instead of ZeroMQ

ZeroMQ has license not compatible with RDKB.

Plan to use nng instead of zmq

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
core-image-minimal.bbappend: add license banner

Each text file in this repo must contain license banner.

Add missed license banner into core-image-minimal.bbappend

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
prplmesh.bb: adopt recipe to build in scope of RDKB for RPi 3

RPi board is easier to buy.
Also RDKB build for Turris Omnia isn't stable and require
many hacks to just be built even without meta-prplmesh

Use prplMesh commit 8a331d9fa1631782cf06603ee612ce6133ee0c1a
and remove workaround for nonstandard name of "mapfcommon.so"
library (without "lib" prefix)

Add 4 patches for prplMesh:
* 001-add_find_nl_for_nl80211_bwl_type.patch
* 002-allow_redefine_beerocks_vars_from_cmdline.patch
* 003_prplmesh_util_stop_use_hard_coded_wlan_iface_names.patch
* 004-get_hostapd_ctrl_iface_pathes_from_conf.patch

Customize WLAN1/WLAN2 ifaces and hostapd pathes
  because RPi has non prplMesh default name and locations of then

  Use PACKAGECONFIG for features which influence not only to
  EXTRA_OECMAKE
  but on DEPENDS too

  Instal CMake find package modules for libnl from prplMesh sources

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
Replace ccsp wifi agent with prplMesh

RDKB CcspWiFiAgent uses nl80211 on RPi and will conflict
with prplMesh

Replace CcspWiFiAgent in RDKB image with prplMesh.

Note: hostapd now will not run automatically because
CcspWiFiAgent was a trigger to run it

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
README.md: add instructions for RDKB RPi 3 B+

Add manual how build RDKB image for RPi 3 B+
and how run prplMesh after board boot
   (prplmesh init script is absent yet
    and out of this task scope)

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

Commit:
prplmesh.bb: remove explicit runtime dependency on "readline" lib

RDEPENDS_${PN} used for runtime dependencies.
In Yocto-community this variable mostly used for dependencies from
plugin to main application or for dependency between applications
(e.g. called with "system()" call). It's uncommon to add shared
library to RDEPENDS explicitely, because all required shared
libraries can be found with shdeps mechanism

Remove "readline" from RDEPENDS_${PN} list

Signed-off-by: Ivan Efimov <i.efimov@inango-systems.com>

VladyslavTupikin commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,108 @@ inherit cmake pythonnative
+BEEROCKS_WLAN1_IFACE_raspberrypi-rdk-broadband = "wlan0"
+BEEROCKS_WLAN2_IFACE_raspberrypi-rdk-broadband = "wlan1"
+BEEROCKS_HOSTAP_WLAN1_CTRL_IFACE_raspberrypi-rdk-broadband="/var/run/hostapd0/wlan0"
+BEEROCKS_HOSTAP_WLAN2_CTRL_IFACE_raspberrypi-rdk-broadband="/var/run/hostapd4/wlan1"
```
```suggestion
BEEROCKS_HOSTAP_WLAN2_CTRL_IFACE_raspberrypi-rdk-broadband="/var/run/hostapd1/wlan1"
```
Why `hostapd4` in the path?

MuteSpirit replied:
It's RPi specific

VladyslavTupikin commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,108 @@ inherit cmake pythonnative
-SRCREV = "911392026280d9961823a9359e6cc4f08c9ee0a2"
-SRC_URI = "git://github.com/prplfoundation/prplMesh.git;protocol=http;branch=master"
+SRCREV = "8a331d9fa1631782cf06603ee612ce6133ee0c1a"
+SRC_URI = "git://github.com/prplfoundation/prplMesh.git;protocol=http;branch=master \
```
nitpick: '\' symbol can be aligned in 1 column.

VladyslavTupikin commented on README.md:
````diff
@@ -5,7 +5,57 @@ This code is subject to the terms of the BSD+Patent license.
+git clone -b rdkb-next https://github.com/prplfoundation/meta-prplmesh
+```
+
+* Source build env
````
Maybe better to create configuration script which will add all necessary changes into the build config (bblayers.conf, etc) ?

VladyslavTupikin reviewed (approved):
Looks good for me.

arnout commented on recipes-connectivity/nng/nng_1.2.5.bb:
```diff
@@ -0,0 +1,25 @@
+SUMMARY = "nanomsg-next-generation -- light-weight brokerless messaging"
```
Note that as of current prplmesh master, nng is no longer needed.

arnout commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,108 @@ inherit cmake pythonnative
+
+SRCREV = "8a331d9fa1631782cf06603ee612ce6133ee0c1a"
+SRC_URI = "git://github.com/prplfoundation/prplMesh.git;protocol=http;branch=master \
+           file://001-add_find_nl_for_nl80211_bwl_type.patch \
+           file://002-allow_redefine_beerocks_vars_from_cmdline.patch \
+           file://003_prplmesh_util_stop_use_hard_coded_wlan_iface_names.patch \
+           file://004-get_hostapd_ctrl_iface_pathes_from_conf.patch"
```
As I mentioned in the RDK-B meeting, we can't do RDK builds as part of prplMesh CI until all of these are upstream.

arnout commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,108 @@ inherit cmake pythonnative
-SRC_URI = "git://github.com/prplfoundation/prplMesh.git;protocol=http;branch=master"
 
-BUILD_TYPE ?= "Release"
+PACKAGECONFIG ??= "${BWL_TYPE} ${MSGLIB}"
```
MSGLIB no longer exists in master.

arnout commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,108 @@ inherit cmake pythonnative
+
+BEEROCKS_BRIDGE_IFACE_rdk ?= "brlan0"
+#
+# FIXME: There is a bug in script printing status
+#        if log files paths are not standard
```
This should also be resolved now, at least on the gateway, since the script no longer uses the log files to check operational status on the gateway.

arnout commented on recipes-connectivity/prplmesh/prplmesh.bb:
```diff
@@ -8,27 +8,108 @@ inherit cmake pythonnative
+EXTRA_OECMAKE += "${@oecmake_var('BEEROCKS_WPA_SUPPLICANT_WLAN2_CTRL_IFACE', d)}"
+EXTRA_OECMAKE += "${@oecmake_var('BEEROCKS_WPA_SUPPLICANT_WLAN3_CTRL_IFACE', d)}"
+
+do_configure_prepend() {
+    install -d ${D}${datadir}/cmake/Modules
+    install -m 0644 ${S}/cmake/Findnl-3.cmake ${D}${datadir}/cmake/Modules/
+    install -m 0644 ${S}/cmake/Findnl-genl-3.cmake ${D}${datadir}/cmake/Modules/
+}
```
Why is this needed? prplmesh CMakeLists.txt includes ${S}/cmake into its search path, so it shouldn't be needed to do this.

MuteSpirit replied:
Will be fixed in https://jira.prplfoundation.org/browse/PPM-282

arnout commented on recipes-connectivity/prplmesh/prplmesh/002-allow_redefine_beerocks_vars_from_cmdline.patch:
```diff
@@ -0,0 +1,166 @@
+commit 9f1aa2d7bfcc74a38ebab39892283468b0c7472d
```
This one was merged upstream.

MuteSpirit replied:
Used prplMesh commit will be updated in separate task https://jira.prplfoundation.org/browse/PPM-281

arnout commented on recipes-connectivity/prplmesh/prplmesh/003_prplmesh_util_stop_use_hard_coded_wlan_iface_names.patch:
```diff
@@ -0,0 +1,52 @@
+commit d090b1ad12036d6339746258c8b7be053f27395e
```
This one was merged upstream.

MuteSpirit replied:
Used prplMesh commit will be updated in separate task https://jira.prplfoundation.org/browse/PPM-281
